packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChartThumbnail.tsx

```diff
@@ -1,27 +1,211 @@
+        const { xAxisConfig } = this.manager
+        const custom = { labelPadding: 2 }
+        return new AxisConfig({ ...custom, ...xAxisConfig }, this)
+    }
+
+    @computed private get yAxisConfig(): AxisConfig {
+        const { yAxisConfig } = this.manager
+        const custom = { labelPadding: 8 }
+        return new AxisConfig({ ...custom, ...yAxisConfig }, this)
+    }
+
+    @computed private get verticalAxisPart(): VerticalAxis {
+        return this.chartState.toVerticalAxis(this.yAxisConfig)
+    }
+
+    @computed private get horizontalAxisPart(): HorizontalAxis {
+        return this.chartState.toHorizontalAxis(this.xAxisConfig)
     }
+
+    @computed private get dualAxis(): DualAxis {
+        return new DualAxis({
+            bounds: this.innerBounds,
+            horizontalAxis: this.horizontalAxisPart,
+            verticalAxis: this.verticalAxisPart,
+        })
+    }
+
+    @computed private get sizeRange(): [number, number] {
+        return toSizeRange(this.chartState, this.innerBounds)
+    }
+
+    @computed get sizeScale(): ScaleLinear<number, number> {
+        return scaleSqrt()
+            .domain(this.chartState.sizeDomain)
+            .range(this.sizeRange)
+    }
+
+    private getPointRadius(value: number | undefined): number {
+        const radius =
+            value !== undefined
+                ? this.sizeScale(value)
+                : this.sizeScale.range()[0]
+
+        // We are enforcing the minimum radius/width just before render,
+        // it should not be enforced earlier than that.
+        return Math.max(
+            radius,
+            this.props.chartState.isConnected
+                ? SCATTER_LINE_MIN_WIDTH
+                : SCATTER_POINT_MIN_RADIUS
+        )
+    }
+
+    override render(): React.ReactElement | null {
+        if (this.chartState.errorInfo.reason) return null
+
+        return (
+            <>
+                <DualAxisComponent
+                    dualAxis={this.dualAxis}
+                    showTickMarks={false}
+                    showEndpointsOnly={true}
+                />
+                <ScatterPointsWithLabels
+                    noDataModalManager={this.manager}
+                    isConnected={this.chartState.isConnected}
+                    hideConnectedScatterLines={
+                        !!this.manager.hideConnectedScatterLines
+                    }
+                    seriesArray={this.chartState.series}
+                    dualAxis={this.dualAxis}
+                    colorScale={
+                        !this.chartState.colorColumn.isMissing
+                            ? this.chartState.colorScale
+                            : undefined
+                    }
+                    sizeScale={this.sizeScale}
+                    baseFontSize={this.fontSize}
+                    focusedSeriesNames={
+                        this.chartState.selectionArray.selectedEntityNames
+                    }
```

Comment:
This doesn't appear to be an issue with the viz code, but entity selections appear to be incorrectly applied here http://staging-site-rich-data-revisions/data?q=Self-reported+happiness+vs.+GDP+per+capita&countries=Germany (Afghanistan is highlighted) or here http://staging-site-rich-data-revisions/data?q=Perceived+importance+of+national+funding+for+anxiety%2Fdepression+research+vs.+GDP+per+capita&countries=Australia (Albania is highlighted)

Reply:
That was a really dumb bug in the search code that is now fixed. Thanks for spotting!